OpenRA.Mods.Common/Orders/PlaceBuildingOrderGenerator.cs

```diff
@@ -210,5 +219,25 @@ public IEnumerable<IRenderable> RenderAfterWorld(WorldRenderer wr, World world)
 		}
 
 		public string GetCursor(World world, CPos xy, MouseInput mi) { return "default"; }
+
+		IEnumerable<Order> TryClearBlockers(World world, CPos topLeft)
+		{
+			var allTiles = FootprintUtils.Tiles(world.Map.Rules, building, buildingInfo, topLeft).ToArray();
+			var neightborTiles = Util.ExpandFootprint(allTiles, true).Except(allTiles)
+				.Where(t => world.Map.Contains(t)).ToList();
+
+			var blockers = allTiles.SelectMany(t => world.ActorMap.GetUnitsAt(t))
+				.Where(a => a.Owner == producer.Owner && a.HasTrait<Mobile>());
+
+			foreach (var blocker in blockers)
+			{
+				var availableCells = neightborTiles.Where(t => blocker.Trait<Mobile>().CanEnterCell(t));
+
+				yield return new Order("Move", blocker, false)
+				{
+					TargetLocation = blocker.ClosestCell(availableCells)
```

Comment:
One issue I ran into was a cell becoming full while actors were still trying to find a destination so they ended up hanging around still.

Of course you can spam click to eventually solve this.